src/main/java/com/sprint/mission/discodeit/service/basic/BasicChannelService.java

```diff
@@ -0,0 +1,589 @@
+                ch.getDescription(),
+                ch.getParticipantIds()
+            ))
+            .toList()
+    );
+
+  }
+
+  /**
+   * - [ ] DTO를 활용해 파라미터를 그룹화합니다. - 수정 대상 객체의 id 파라미터, 수정할 값 파라미터 - [ ] PRIVATE 채널은 수정할 수 없습니다.
+   */
+  @Override
+  public ChannelUpdateResponse updateChannel(String channelName, ChannelUpdateRequest request) {
+    Channel channel = channelRepository.findChannelByName(channelName);
+    User owner = userRepository.findUserByName(request.ownerName());
+
+    channel.updateChannelName(request.channelName());
+    channel.setChannelType(request.channelType());
+    channel.updateOwnerUser(owner);
+
+    if (channel.getChannelType().equals(CHANNEL_TYPE_PRI)) {
+      throw new IllegalChannelException("PRIVATE 채널은 수정할 수 없습니다.");
+    }
+
+    channelRepository.saveChannel(channel);
+
+    return new ChannelUpdateResponse(channel.getId(), channel.getChannelName(),
+        channel.getChannelOwnerUser().getId(), channel.getChannelType());
+  }
+
+
+  /**
+   * 채널에 메세지 추가
+   */
+  @Override
+  public void sendMessage(ChannelAddMessageRequest request) {
+    Channel findChannel = channelRepository.findChannelById(request.channelId());
+    MessageResponse message = messageService.getMessageById(request.messageId());
+
+    if (findChannel == null || message == null) {
+      throw new ChannelNotFoundException(request.channelId());
+    }
+
+    User sender = userRepository.findUserById(message.senderId());
+    User receiver = userRepository.findUserById(message.receiverId());
+
+    Message createdMessage = new Message(message.messageId(), message.messageTitle(),
+        message.messageContent(), sender, receiver);
+    findChannel.addMessageInChannel(createdMessage);
+    channelRepository.saveChannel(findChannel);
+
+  }
+
+  /**
+   * 관련된 도메인도 같이 삭제합니다. - `Message`, `ReadStatus`
+   */
+  @Override
+  public UUID removeChannelById(UUID channelUUID) {
+    Channel findChannel = channelRepository.findChannelById(channelUUID); // 여기서 채널을 못찾음
+
+    Map<UUID, Message> messageMap = Optional.ofNullable(findChannel.getChannelMessages())
+        .orElseThrow(MessageNotFoundException::new);
+
+    // 채널 내 메시지 삭제 (MessageService 사용)
+    if (!messageMap.isEmpty()) {
+      List<UUID> messageIds = new ArrayList<>(messageMap.keySet());
+      for (UUID messageId : messageIds) {
+        messageService.deleteMessage(String.valueOf(messageId));
+      }
+    }
+
+    // 4. ReadStatus 삭제 - 새로운 리스트에 삭제할 ID들을 모은 후 한꺼번에 처리
+    Map<UUID, ReadStatus> allReadStatuses = readStatusRepository.findAll();
+
+    if (!allReadStatuses.isEmpty()) {
+      List<UUID> readStatusesToRemove = new ArrayList<>();
+
+      for (ReadStatus readStatus : allReadStatuses.values()) {
+        if (readStatus.getChannelId().equals(channelUUID)) {
+          readStatusesToRemove.add(readStatus.getChannelId());
+        }
+      }
+
+      for (UUID statusId : readStatusesToRemove) {
+        readStatusRepository.remove(statusId);
+      }
```

Comment:
전부 가져오지 말고 채널에 해당하는 ReadStatus만 가져오는건 어때요?

```suggestion
    // 4. ReadStatus 삭제 - 새로운 리스트에 삭제할 ID들을 모은 후 한꺼번에 처리
    Map<UUID, ReadStatus> allReadStatuses = readStatusRepository.findByChannelId(channelUUID);

    if (!allReadStatuses.isEmpty()) {
      List<UUID> readStatusesToRemove = new ArrayList<>();

      for (ReadStatus readStatus : allReadStatuses.values()) {
        if (readStatus.getChannelId().equals(channelUUID)) {
          readStatusesToRemove.add(readStatus.getChannelId());
        }
      }

      for (UUID statusId : readStatusesToRemove) {
        readStatusRepository.remove(statusId);
      }
```